src/jit/rangecheck.cpp

```diff
@@ -311,18 +312,27 @@ void RangeCheck::Widen(BasicBlock* block, GenTreePtr stmt, GenTreePtr tree, Sear
     if (range.LowerLimit().IsDependent() || range.LowerLimit().IsUnknown())
     {
         // To determine the lower bound, ask if the loop increases monotonically.
-        bool increasing = IsMonotonicallyIncreasing(tree, path);
+        bool increasing = IsMonotonicallyIncreasing(tree);
         JITDUMP("IsMonotonicallyIncreasing %d", increasing);
         if (increasing)
         {
             GetRangeMap()->RemoveAll();
-            *pRange = GetRange(block, stmt, tree, path, true DEBUGARG(0));
+            *pRange = GetRange(block, tree, true DEBUGARG(0));
         }
     }
 }
 
-bool RangeCheck::IsBinOpMonotonicallyIncreasing(GenTreePtr op1, GenTreePtr op2, genTreeOps oper, SearchPath* path)
+bool RangeCheck::IsBinOpMonotonicallyIncreasing(GenTreeOp* binop)
 {
+#ifdef LEGACY_BACKEND
+    assert(binop->OperIs(GT_ADD, GT_ASG_ADD));
```

Comment:
Can you simplify this to assert `OperIs(GT_ADD)` in all builds, then?

Reply:
No, now `IsBinOpMonotonicallyIncreasing` gets a node that can be `GT_ADD` or `GT_ASG_ADD` instead of caller lying to it that it is always `GT_ADD`.

Reply:
Ok, I'm following now, thanks.  Looks good, will merge when CI completes.

Reply:
> Looks good, will merge when CI completes.

Ha ha, that won't happen soon, Perf Build and Test takes ages.

Do you want me to squash?

Reply:
> Do you want me to squash?

Up to you, we don't seem to have a consistent convention here across the team/project.